product-service/src/postProduct.ts

```diff
@@ -0,0 +1,100 @@
+import { APIGatewayProxyHandler } from "aws-lambda";
+import "source-map-support/register";
+import { Client } from "pg";
+
+import { dbOptions } from './dbOptions'
+
+const headers = {
+  "Access-Control-Allow-Headers": "Content-Type",
+  "Access-Control-Allow-Origin": "*",
+  "Access-Control-Allow-Methods": "OPTIONS,POST",
+}
+
+export const postProduct: APIGatewayProxyHandler = async (event, _context) => {
+  console.log('Lambda function: postProduct.  Event:', event);
+  let parsedBody;
+  try{
+    parsedBody = JSON.parse(
+      event.body
+    );
+  } catch (err){
+    return {
+      statusCode:400,
+      headers,
+      body: JSON.stringify({
+        message: "body is invalid or missing",
+      }, null, 2),
+    };
+  }
+  const { title, description = "", price = 0, count = 0 } = parsedBody;
+
+  let statusCode = 404;
+  let result = {};
+
+  if (!title || typeof title !== 'string') {
+    statusCode = 400;
+    result = {
+      message: "product data is invalid: title is missing or wrong type",
+    };
+  } else if (!price || isNaN(parseInt(price)) ) {
+    statusCode = 400;
+    result = {
+      message: "product data is invalid: price is incorrect",
+    };
```

Comment:
I think, better to use third-party validator library, because it's hard to catch all invalid inputs this way. Try to send 99999999999999999999999999999999999 in the count or price field, and you will get 500 instead of 400. Currently we can send it by FE or by API call.

Reply:
I've replaced custom validation with joi validator [f476c32 ](https://github.com/YakovGrischenko/nodejs-aws-be/pull/2/commits/f476c3204ff8d8b30c307781593b4e225f19efc2)